packages/field/src/FormControlMixin.js

```diff
@@ -76,7 +76,31 @@ export const FormControlMixin = dedupeMixin(
       }
 
       get inputElement() {
-        return (this.$$slot && this.$$slot('input')) || this.querySelector('[slot=input]'); // eslint-disable-line
+        if (!this.__inputElementCache) {
+          this.__inputElementCache = [...this.children].find(el => el.slot === 'input');
+        }
+        return this.__inputElementCache;
+      }
+
+      get labelElement() {
+        if (!this.__labelElementCache) {
+          this.__labelElementCache = [...this.children].find(el => el.slot === 'label');
+        }
+        return this.__labelElementCache;
+      }
+
+      get helpTextElement() {
+        if (!this.__helpTextElementCache) {
+          this.__helpTextElementCache = [...this.children].find(el => el.slot === 'help-text');
+        }
+        return this.__helpTextElementCache;
+      }
+
+      get feedbackElement() {
+        if (!this.__feedbackElementCache) {
+          this.__feedbackElementCache = [...this.children].find(el => el.slot === 'feedback');
+        }
+        return this.__feedbackElementCache;
```

Comment:
some thoughts

1. why remove $$slot helper which does the caching and reimplement it everythere? same situation as with other helpers/mixins we are removing for some reason
2. even if we remove it, I would not bother and do this optimization, it hardly effects the performance, or if it will, we will make optimisations later
3. and most important: why not just `this.querySelector('[slot="my-slot-name"]')`?